telegram/request/_httpxrequest.py

```diff
@@ -0,0 +1,188 @@
+#
+#  A library that provides a Python interface to the Telegram Bot API
+#  Copyright (C) 2015-2022
+#  Leandro Toledo de Souza <devs@python-telegram-bot.org>
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser Public License
+#  along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains methods to make POST and GET requests using the httpx library."""
+import logging
+from typing import Tuple, Optional
+
+import httpx
+
+from telegram._utils.defaultvalue import DefaultValue
+from telegram._utils.types import ODVInput
+from telegram.error import TimedOut, NetworkError
+from telegram.request import BaseRequest, RequestData
+
+
+# Note to future devs:
+# Proxies are currently only tested manually. The httpx development docs have a nice guide on that:
+# https://www.python-httpx.org/contributing/#development-proxy-setup (also saved on archive.org)
+# That also works with socks5. Just pass `--mode socks5` to mitmproxy
+
+_logger = logging.getLogger(__name__)
+
+
+class HTTPXRequest(BaseRequest):
+    """Implementation of :class:`~telegram.request.BaseRequest` using the library
+    `httpx <https://www.python-httpx.org>`_.
+
+    Args:
+        connection_pool_size (:obj:`int`, optional): Number of connections to keep in the
+            connection pool. Defaults to ``1``.
+
+            Note:
+                Independent of the value, one additional connection will be reserved for
+                :meth:`telegram.Bot.get_updates`.
+        proxy_url (:obj:`str`, optional): The URL to the proxy server. For example
+            ``'http://127.0.0.1:3128'`` or ``'socks5://127.0.0.1:3128'``. Defaults to :obj:`None`.
```

Comment:
I briefly discussed this with noam. IIRC the punch line was that we only make requests to `api.telegram.org` so having different proxies is unlikely to be of actual use. If people want it at some point, we can ofc still reconsider. what do you think?

Reply:
ah good point, we can leave as is then. P.S. I suggested this because I was using `context.bot._request[-1]._client` for making requests to other websites (I was lazy to make another httpx instance) and thought that maybe the proxy shouldn't apply to  another site